### PR TITLE
feat: stream build output and print build/rebuild info in table

### PR DESCRIPTION
**Pull request by tdejager**
Two additions:

* Brings back streaming of `rattler-build` output
* Prints `Build`, `Rebuild` information nicely.

### CODE REVIEW

File: src/repror/internals/commands.py [hunk 1]
```diff
@@ -42,24 +42,24 @@ def run_streaming_command(
...
-    with subprocess.Popen(command, cwd=cwd, env=env, **kwargs) as process:
-        # Print stderr as it comes in
-        for line in io.TextIOWrapper(
-            process.stderr, encoding="utf-8"
-        ):  # or another encoding
-            print(line, end="")
+    with subprocess.Popen(args=command, cwd=cwd, env=env, **kwargs) as process:
+        main_stream = process.stderr if stream_type.is_stderr else process.stdout
+        if main_stream:
+            # Print stderr as it comes in
```
Comment (nichmor) [inline src/repror/internals/commands.py:53]: `# Print stderr or stdout as it comes in`

File: src/repror/internals/commands.py [hunk 2]
```diff
@@ -42,24 +42,24 @@ def run_streaming_command(
...
         kwargs = {"stdout": subprocess.PIPE}
 
-    with subprocess.Popen(command, cwd=cwd, env=env, **kwargs) as process:
-        # Print stderr as it comes in
-        for line in io.TextIOWrapper(
-            process.stderr, encoding="utf-8"
-        ):  # or another encoding
-            print(line, end="")
+    with subprocess.Popen(args=command, cwd=cwd, env=env, **kwargs) as process:
+        main_stream = process.stderr if stream_type.is_stderr else process.stdout
```
Comment (nichmor) [inline src/repror/internals/commands.py:51]: let write it :
`main_stream, other_stream = process.stderr, process.stdout if ... else ...`

File: src/repror/internals/commands.py [hunk 3]
```diff
@@ -42,24 +42,24 @@ def run_streaming_command(
...
+                main_stream, encoding="utf-8"
+            ):  # or another encoding
+                print(line, end="")
 
         # Also print the other output if there is any
-
-        out = process.stdout if stream_type.is_stderr else process.stderr
-        if out:
-            print(process.stdout.readline())
+        other_stream = process.stdout if stream_type.is_stderr else process.stderr
```
Comment (nichmor) [inline src/repror/internals/commands.py:60]: so we can remove it because we already untupled it ^

**Review by nichmor — APPROVED** [review]
only small question:
does it still persist the error when throwing exception?

**Comment by tdejager** [comment]
> only small question: does it still persist the error when throwing exception?

This works, again!

**Pull request merged**